ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -174,7 +174,25 @@ data InitLog blk =
     -- We record the reason why it was skipped.
     --
     -- NOTE: We should /only/ see this if data corrupted occurred.
-  | InitFailure DiskSnapshot (InitFailure blk) (InitLog blk)
+  | InitFailure DiskSnapshot (RecoverableInitFailure blk) (InitLog blk)
+  deriving (Show, Eq, Generic)
+
+-- | Errors that can happen during initialization and cannot be recovered,
+-- leading to a call to @error@ and aborting the program.
+data FatalInitFailure blk = InitFailureTooRecent (RealPoint blk)
```

Comment:
`streamAll` is only called once. And this constructor only occurs once, at that call. So you could just inline it into `streamAll` and avoid the "noise" of this data type.

Reply:
I just inlined the call to `error` into the `streamAll` call as you suggested.